caikit/core/model_management/local_job_predictor.py

```diff
@@ -0,0 +1,221 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+The LocalJobPredictor uses a local thread to launch and manage each prediction job
+
+model_management:
+    job_predictors:
+        <predictor name>:
+            type: LOCAL
+            config:
+                # ! Inherits config from LocalJobBase
+                # Path to a local directory that holds the results. Defaults
+                # to a temporary directory
+                result_dir: <null or str>
+"""
+
+# Standard
+from datetime import datetime
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import Any, Dict, Optional
+import shutil
+
+# First Party
+import aconfig
+import alog
+
+# Local
+from ..data_model import DataObjectBase
+from ..exceptions import error_handler
+from ..modules import ModuleBase
+from .job_predictor_base import JobPredictorBase
+from .local_job_base import LocalJobBase
+from caikit.core.exceptions.caikit_core_exception import (
+    CaikitCoreException,
+    CaikitCoreStatusCode,
+)
+
+log = alog.use_channel("LOC-TRNR")
+error = error_handler.get(log)
+
+# Constant for the result file name. This is exposed here to better support
+# subclassing
+RESULT_FILE_NAME = "result.bin"
```

Comment:
Is there a reason this isn't configurable? Seems like a user may want to be able to avoid overwriting and/or name things to fit their needs.

Reply:
So this is just the sub file name inside a unque folder for that id e.g. `<result_dir>/<future id>/result.bin`. Initially I had `result.bin` hard coded in the `__init__` but I moved it to a constant for better subclassing. I don't think it needs to be user configurable since the `result_dir` already is but will defer to your judgement. This is a low CA for me